### DISCUSSION TITLE
"dynamic" dependency group names

### DISCUSSION THREAD

**Discussion by gizero** [renovatebot/renovate#31105] [Request Help] [answered]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

GitLab, renovate v38.x

### Please tell us more about your question or problem

Hi! Hope someone can steer me in the right direction with this...
I'm trying to figure out if there's a way to configure a package rule in order to group "twin" dependencies in groups. Those groups should be named after the `<depX>` with `<depX>-prod` and `<depX>-preprod` being grouped together. Nowadays I'm already able to get two separate MRs for the above mentioned dependencies. What I'd want to avoid is to be required to have a package rule for each possible value of `<depX>`, which would solve the problem but badly scale.

I was thinking about being able to use capture groups within `matchXXX` package rules and then referencing captured values from the `groupName` field, but it seems it's not documented as a valid config.

Is there any already supported way to achieve this goal? I couldn't find any while going through the documentation on grouping.

Thank you in advance for any response or suggestion.

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by rarkins** [comment, marked as answer]
Use the handlebars `replace` directive in templating to strip `-prod` and `-preprod` from `depName` in `commitMessageTopic`. Then they'll all be in branches like `renovate/depX-1.x`

I know it's complex but what you're asking for is too..

**Reply by gizero** [reply]
Thanks @rarkins for the valuable advice. I was able to achieve the goal! One thing I had to figure out was that I also had to tune `branchTopic` to get the grouping right. I'm sharing the config snippet here, in case you can suggest further improvements.

```
  "packageRules": [
    {
      "description": "ensure prod and preprod variants of each dependency get grouped together",
      "matchDepNames": [
        "/^.*-(?:pre)?prod$/"
      ],
      "branchTopic": "{{{replace '-(?:pre)?prod$' '' depNameSanitized}}}-{{{newMajor}}}{{#if separateMinorPatch}}{{#if isPatch}}.{{{newMinor}}}{{/if}}{{/if}}.x{{#if isLockfileUpdate}}-lockfile{{/if}}",
      "commitMessageTopic": "{{{replace '-(?:pre)?prod$' '' depName}}}"
    }
 ]
```

Thanks again for helping out! 🚀
